clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -614,14 +620,27 @@ static std::string convertCmakeCommandsToLower(const std::string &InputString) {
       std::string Str = Line.substr(Begin, End - Begin);
       std::transform(Str.begin(), Str.end(), Str.begin(),
                      [](unsigned char Char) { return std::tolower(Char); });
-      if (cmake_commands.find(Str) != cmake_commands.end()) {
+      auto Iter = cmake_commands.find(Str);
+      if (Iter != cmake_commands.end()) {
         for (int Idx = Begin; Idx < End; Idx++) {
           Line[Idx] = Str[Idx - Begin];
         }
+        if (!Iter->second) {
+          std::string Prefix =
+              FileName + ":" + std::to_string(Count) + ":warning:";
+          std::string WaringContent =
+              "DPCT2000: migration of syntax \"" + Str +
```

Comment:
Is it better to move the message to:
```
//===--------------- DiagnosticsBuildScript.inc ---------------------------===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//

// FORMAT: {NAME, ID, MESSAGE}
DEF_COMMENT(GEN_MAKEFILE_LIB, 2001, "You can link with more library by add them here.")
DEF_COMMENT(GEN_MAKEFILE_OPT, 2002, "You can change different compiler optimization options here.")
```
?

Reply:
good advice, will be updated it..